runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useQuery.ts

```diff
@@ -1,11 +1,26 @@
-import { useMemo } from 'react';
-
 import { useRouter } from 'next/router';
+
 const useQuery = () => {
     const { query } = useRouter();
-    const search = useMemo(() => query.search, [query.search]);
 
-    return useMemo(() => new URLSearchParams(search as string), [search]);
+    const firstValueFrom = (param: string) => {
+        const valueOfQueryParam = query[param];
+
+        if (Array.isArray(valueOfQueryParam)) {
+            return valueOfQueryParam[0];
+        }
+
+        if (typeof valueOfQueryParam === 'string') {
+            return valueOfQueryParam;
+        }
+
+        return valueOfQueryParam;
```

Comment:
avoid linking words like `of` in this case. See how simple words rearrange can do the trick `queryParam` or `queryParamValue`. Note that we expect under `queryParam` value of it, if it differs then you should adjust the name